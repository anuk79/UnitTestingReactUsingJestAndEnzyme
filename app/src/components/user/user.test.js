
// vendors
import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import { debounce } from 'throttle-debounce';

// components
import UserView from './user';

jest.mock('throttle-debounce', () => {
    return {    
        debounce: jest.fn()
    }
});

describe("UserView component", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<UserView 
                userDetails={{ firstName: 'test firstName', lastName: 'test lastName'}}
            />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();

        expect(debounce).toBeCalled();
        expect(wrapper.find('h3').length).toBe(1);
        expect(wrapper.find('div').length).toBe(2);
        expect(wrapper.find('span').length).toBe(2);
        expect(wrapper.find('label').length).toBe(2);
        expect(wrapper.find('h3').text()).toEqual('User Details:');
        expect(wrapper.find('span').at(0).text()).toEqual('test firstName');
        expect(wrapper.find('span').at(1).text()).toEqual('test lastName');
    });
});