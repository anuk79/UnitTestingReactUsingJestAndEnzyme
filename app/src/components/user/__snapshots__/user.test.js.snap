// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView component should render correctly 1`] = `
<Fragment>
  <h3>
    User Details:
  </h3>
  <div>
    <label>
      First Name: 
    </label>
    <span>
      test firstName
    </span>
  </div>
  <div>
    <label>
      Last Name: 
    </label>
    <span>
      test lastName
    </span>
  </div>
</Fragment>
`;
